frontend/src/components/pages/CampOverview/PickupDropoffTableRow.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+
+import { Text, Tr, Td, HStack } from "@chakra-ui/react";
+
+import { ReactComponent as SunriseIcon } from "../../../assets/icon_sunrise.svg";
+import { ReactComponent as SunsetIcon } from "../../../assets/icon_sunset.svg";
+
+const weekdayMap: { [key: number]: string } = {
+  1: "Monday",
+  2: "Tuesday",
+  3: "Wednesday",
+  4: "Thursday",
+  5: "Friday",
+};
+
+// findTime consumes an array of dates and a specified weekday (i.e. Sunday = 0, Monday = 1, etc.).
+// Returns the time of the first date in the array that matches the weekday (i.e. 8:00 AM) or XX:XX if none found.
+const findTime = (dates: Date[] | undefined, weekday: number) => {
+  let finalTime = "XX:XX";
+
+  if (dates) {
+    dates.forEach((date) => {
```

Comment:
It seems like you want to find the date in the list of dates where the day matches the current day you're on. 

This is fine as is, but maybe using the `array.find()` function would make this more clear? Then if you don't find the weekday, you can return `XX:XX`, otherwise you have a filtered date to do the rest. of the processing. This should be fine since only 1 date in the dates array should match the given day.